crates/driver/openapi.yml

```diff
@@ -294,29 +294,63 @@ components:
       example: "0x30cff40d9f60caa68a37f0ee73253ad6ad72b45580c945fe3ab67596476937197854163b1b0d24e77dca702b97b5cc33e0f83dcb626122a6"
     QuoteResponse:
       oneOf:
-        - description: |
-            Successful Quote
+        - $ref: "#/components/schemas/LegacyQuote"
+        - $ref: "#/components/schemas/QuoteWithJitOrders"
+        - $ref: "#/components/schemas/Error"
+    LegacyQuote:
+      description: |
+        Successful Quote
 
-            The Solver knows how to fill the request with these parameters.
+        The Solver knows how to fill the request with these parameters.
+
+        If the request was of type `buy` then the response's buy amount has the same value as the
+        request's amount and the sell amount was filled in by the server. Vice versa for type
+        `sell`.
+      type: object
+      properties:
+        amount:
+          $ref: "#/components/schemas/TokenAmount"
+        interactions:
+          type: array
+          items:
+            $ref: "#/components/schemas/Interaction"
+        solver:
+          description: The address of the solver that quoted this order.
+          $ref: "#/components/schemas/Address"
+        gas:
+          type: integer
+          description: How many units of gas this trade is estimated to cost.
+    QuoteWithJitOrders:
+      description: |
+        Successful Quote with JIT orders support.
 
-            If the request was of type `buy` then the response's buy amount has the same value as the
-            request's amount and the sell amount was filled in by the server. Vice versa for type
-            `sell`.
+        The Solver knows how to fill the request with these parameters.
+      type: object
+      properties:
+        clearingPrices:
+          description: |
+            Mapping of hex token address to the uniform clearing price.
           type: object
-          properties:
-            amount:
-              $ref: "#/components/schemas/TokenAmount"
-            interactions:
-              type: array
-              items:
-                $ref: "#/components/schemas/Interaction"
-            solver:
-              description: The address of the solver that quoted this order.
-              $ref: "#/components/schemas/Address"
-            gas:
-              type: integer
-              description: How many units of gas this trade is estimated to cost.
-        - $ref: "#/components/schemas/Error"
+          additionalProperties:
+            $ref: "#/components/schemas/BigUint"
+        preInteractions:
+          type: array
+          items:
+            $ref: "#/components/schemas/Interaction"
+        interactions:
+          type: array
+          items:
+            $ref: "#/components/schemas/Interaction"
+        solver:
```

Comment:
I think we still need to have the optional `txOrigin` (also not present in the legacy endpoint spec) to allow solvers to provide custom addresses their solutions should get simulated from.

Reply:
Thanks, it was missing in the original schema. Added.